RFC/src/RFC-0310_AssetTemplates.md

```diff
@@ -0,0 +1,218 @@
+| name                               | `string[64]`  | The name or identifier for the asset. See [Name and Description](#name-and-description)                                       |
+| description                        | `string[128]` | A short description of the asset - fits in a tweet. . See [Name and Description](#name-and-description)                       |
+| tldi                               | `string[16]`  | The Top Level Digital Issuer (TLDI) for the asset. The TLDI must be owned by the asset issuer, or empty / "Other" for no TLDI |
+| tldi_signature                     | `u256`        | Digital signature for the given TLDI                                                                                          |
+| template_number                    | `u64`         | The template number and version for this asset (e.g. 0x01 => Simple single-use token)                                         |
+| asset_expiry                       | `u64`         | A timestamp or block height after which the asset will automatically expire. Zero for arbitrarily long-lived assets           |
+| **Validation Committee selection** |               |                                                                                                                               |
+| committee_mode                     | `u8`          | The validation committee mode, either `PERMISSIONED` (0) or `PERMISSIONLESS` (1)                                              |
+| committee_parameters               | Object        |                                                                                                                               |
+| fee                                | `u64`         | The fee the issuer is paying, in nanoTari for the asset creation instruction                                                  |
+| commitment                         | `u256`        | A time-locked commitment for the fee amount                                                                                   |
+| commitment_sig                     | `u256`        | A signature proving the issuer is able to spend the commitment                                                                |
+| initial_state_hash                 | `u256`        | The hash of the canonical serialisation of the initial template state (of the template-specific data)                         |
+| **Signatures**                     |               |                                                                                                                               |
+| creator_sig                        | `u256`        | A digital signature of the preceding 2 sections’ data using the asset creator’s private key corresponding to the `issuer` PKH |
+| **template-specific data**         |               | Depending on the template, the data in the final section varies and corresponds to the initial asset state                    |
+
```

Comment:
```suggestion
| token_amount                       |  u64          | The amount of tokens in the asset                                                                                             |
| initial_asset_state                |  Object       | The initial asset state, including that for all tokens                                                                        |
| asset_lock_height                  |  u64          | Base node block height at which the asset will start to be "live"                                                             |
```

Reply:
The token supply is template-dependent, so shouldn't be included here

Reply:
I do not follow the reasoning here...
Surely the asset issuer must specify the amount of tokens?

Reply:
Yes, but some templates might not have tokens at all, so it goes in the template-specific metadata